instrumentation/undertow-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/undertow/UndertowServerDispatchTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.undertow;
+
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.CAPTURE_HEADERS;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.ERROR;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.INDEXED_CHILD;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.QUERY_PARAM;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.REDIRECT;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.SUCCESS;
+
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.util.ThrowingRunnable;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.undertow.Handlers;
+import io.undertow.Undertow;
+import io.undertow.util.Headers;
+import io.undertow.util.HttpString;
+import io.undertow.util.StatusCodes;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class UndertowServerDispatchTest extends AbstractHttpServerTest<Undertow> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
+
+  @Override
+  public Undertow setupServer() {
+    Undertow server =
+        Undertow.builder()
+            .addHttpListener(port, "localhost")
+            .setHandler(
+                Handlers.path()
+                    .addExactPath(
+                        SUCCESS.rawPath(),
+                        exchange ->
+                            exchange.dispatch(
+                                k ->
+                                    testing.runWithSpan(
+                                        "controller",
+                                        (ThrowingRunnable<Exception>)
+                                            () -> k.getResponseSender().send(SUCCESS.getBody()))))
```

Comment:
the java AbstractHttpServerTest [has a helper method for this](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpServerTest.java#L118), similar to the ones in groovy

I think it would look like:

```suggestion
                    .addExactPath(
                        SUCCESS.rawPath(),
                        exchange -> controller(SUCCESS, () -> {
                          exchange.getResponseSender().send(SUCCESS.getBody());
                          return null;
                        }))
```

Reply:
i tries this in my second commits https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12207/commits/98c5f088410761a4ee8eded5468573965cd94bb6#diff-b3a26e2313b3c2d5656873760131d5794cccfdda0263105ad63b83527d623981R47

But the test case failed. because the span name in grooy ut is "controller",  `controller(SUCCESS, ` span name is success....

Reply:
done